InnerEye/ML/SSL/lightning_modules/ssl_online_evaluator.py

```diff
@@ -112,16 +131,20 @@ def on_validation_batch_end(self, trainer: pl.Trainer,
         ids_linear_head = tuple(batch[SSLDataModuleType.LINEAR_HEAD][0].tolist())
         if ids_linear_head not in self.visited_ids:
             self.visited_ids.add(ids_linear_head)
+            old_mode = self.non_linear_evaluator.training
+            self.non_linear_evaluator.eval()
             loss = self.shared_step(batch, pl_module, is_training=False)
-            pl_module.log('ssl_online_evaluator/val/loss', loss, on_step=False, on_epoch=True, sync_dist=False)
+            log_on_epoch(pl_module, 'ssl_online_evaluator/val/loss', loss)
             for metric in self.val_metrics:
-                pl_module.log(f"ssl_online_evaluator/val/{metric.name}", metric, on_epoch=True,
-                              on_step=False)  # type: ignore
+                log_on_epoch(pl_module, f"ssl_online_evaluator/val/{metric.name}", metric)
+            self.non_linear_evaluator.train(old_mode)
```

Comment:
Can we update the docstring to include that the train method is called here?

Reply:
The method name may be confusing, this puts the module into either train or eval mode. I've added comments to clarify what this does.